File: medium/0128_longest_consecutive_sequence/solution.cpp
#include <unordered_set>
#include <vector>

class Solution {
public:
  int longestConsecutive(std::vector<int> &nums) {
    std::unordered_set<int> numsSet(nums.begin(), nums.end());

    int maxLen = 0;

    for (int num : numsSet) {
      if (numsSet.find(num - 1) != numsSet.end())
        continue;

      int currLen = 1;

      while (numsSet.find(num + 1) != numsSet.end()) {
        ++num;
        ++currLen;
      }

      maxLen = std::max(maxLen, currLen);
    }

    return maxLen;
  }
};

File: medium/0128_longest_consecutive_sequence/solution.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numsSet = set(nums)

        maxLen = 0

        for num in numsSet:
            if num - 1 in numsSet:
                continue

            currLen = 1

            while num + 1 in numsSet:
                num += 1
                currLen += 1

            maxLen = max(maxLen, currLen)

        return maxLen
